source/slang/slang-check-expr.cpp

```diff
@@ -1368,6 +1385,13 @@ namespace Slang
                 });
             return;
         }
+
+        if (auto typePack = as<ConcreteTypePack>(type))
```

Comment:
Yong's note: the logic here is to make sure we register types inside a type pack to the differentiable type dictionary of a differentiable function.